Explore RNA seq read count data
===============================

> To knit .rmd file, read data files in using "../data"  
> To run chunks in Rstudio, read data files in using "./data"

## Load data and required libraries
Load RNA-seq data and the experimental design files:
```{r}
rDat <- read.table("../data/laml.rnaseq.179_v1.0_gaf2.0_read_count_matrix.txt.tcgaID.txt.gz", sep = "\t", header = TRUE, row.names = 1)
rDes <- read.delim("../data/experimental_design_cleaned.txt")
```

```{r warning=FALSE, message=FALSE}
library(reshape2) # For reshaping data from wide to tall format
library(ggplot2) # for plotting
library(RColorBrewer)
library(plyr)
library(limma)
library(edgeR)
library(VennDiagram)  # for `venn.plot` function
```


## Data inspection
```{r}
str(rDat, max.level = 0)
rDat[1:4, 1:4]
head(names(rDat))
head(rownames(rDat), n = 10)
tail(rownames(rDat), n = 10)
str(rDes, max.level = 0)
head(rDes)
```

RNA-seq data: there are `r nrow(rDat)` transcripts (rows) for `r length(rDat)` patients (columns). The row names are strange, I will attempt to change them.

Experimental design: there are `r nrow(rDes)` rows, representing information for each of the patients with RNA-seq data in the AML TCGA data set, and `r length(rDat)` variables. Note the sample ID naming scheme does not match across `rDat` and `rDes`, so I will need to fix this.


## Clean rDat
Change sample names in `rDat` to match `rDes` by extracting substrings, i.e. extract the numbers in each sample name:
```{r}
names(rDat) <- regmatches(names(rDat), regexpr("(?<=AB.).*",
                                                  names(rDat),
                                                  perl = TRUE))
head(names(rDat))
```

Reorder the columns (patients) to match the order of the experimental design file 'rDes$TCGA\_patient_id`:
```{r}
rDat <- rDat[ , order(names(rDat))]
identical(names(rDat), as.character(rDes$TCGA_patient_id))
```

Now to fix the row names: some row names begin with "?", and each row name has the suffix "|", followed by an integer, then "_calculated". This naming scheme is explained [here](https://wiki.nci.nih.gov/display/TCGA/RNASeq+Data+Format+Specification), where the row names are: "valid HUGO name|GENEID (if no HUGO name, '?' is present)".

Remove the rows with "?" in the row name:
```{r}
# "?" only present at the start of the row name:
# rownames(rDat)[grep("[?]", rownames(rDat))]
length(grep("[?]", rownames(rDat)))
rDat <- rDat[grep("[?]", rownames(rDat), invert = TRUE), ]
dim(rDat)
```


## Filtering
Remove transcripts with read count = 0 across all samples:
```{r}
# Number of transcripts with read count = 0 for all samples
nrow(rDat[rowSums(rDat) == 0, ])
# Remove these transcripts
rDat <- rDat[rowSums(rDat) != 0, ]
dim(rDat)
```

I have decided not to apply any additional filters to our data, since I believe we may be removing biologically significant data. We are working with RNA-seq data from AML patients, a cancer type that is prone to translocations and copy number changes. Therefore, genes with 0 read count values may be instances where genes are completely deleted from the genome and thus no transcription can occur.


## Density plot
Check the density plot of read count values across all samples:
```{r}
rDatMelt <- melt(rDat, variable.name = "Sample", 
                 value.name = "ReadCount")
head(rDatMelt)
ggplot(rDatMelt, aes(ReadCount)) +
  geom_density()
```

The data has to be log transformed:
```{r}
ggplot(rDatMelt, aes(log(ReadCount))) +
  geom_density()
```

A lot of genes have read count values < 1 and become negative values post-log2 transformation.  Therefore, I will add 1 to all values in `rDat`:
```{r}
rDat <- rDat + 1
```

Now re-make the density plot:
```{r fig.align='center'}
rDat <- rDat + 1
rDatMelt <- melt(rDat, variable.name = "Sample", 
                 value.name = "ReadCount")
ggplot(rDatMelt, aes(log2(ReadCount))) +
  geom_density()  
```


## Differential expression analysis using voom

I will use `voom` to perform differential expression analysis. From my experience, `voom` makes the most stringent calls for differential expression. 

**Sex**  
First I will do a simple differential expression analysis: which genes are differentially expressed between males and females?
```{r}
sex <- rDes$Sex
table(sex)
```

Apply scale normalization:
```{r}
normFactor <- calcNormFactors(rDat)
```

Use `voom` to convert read count to log2-ReadCount ready for linear modelling:
```{r}
design <- model.matrix(~ sex)
# The intercept represents females
head(design)
rDatVoom <- voom(rDat, design, plot = TRUE,  
                 lib.size = colSums(rDat) * normFactor)
```

Now find genes differentially expressed between males and females:
```{r}
fit <- lmFit(rDatVoom, design)
fit <- eBayes(fit)
voomSex <- topTable(fit, coef = "sexM", p.value = 1e-5, n = Inf)
nrow(voomSex)
head(voomSex, n = 10)
(voomSexgenes <- rownames(voomSex))
# Lower the FDR threshold: how many genes do we find?
nrow(topTable(fit, coef = "sexM", p.value = 1e-2, n = Inf))
nrow(topTable(fit, coef = "sexM", p.value = 0.01, n = Inf))
# None! Really?
head(topTable(fit, coef = "sexM", n = Inf))
tail(topTable(fit, coef = "sexM", n = Inf))
```
There are no genes differentially expressed between males and females!

**Cytogenetic risk**  
Now to explore another variable, "Cytogenetic_risk". 
```{r}
levels(rDes$Cytogenetic_risk)
table(rDes$Cytogenetic_risk)
```

Which transcripts are differentially expressed between "Good", "Intermediate", and "Poor" cytogenetic risk?

First, remove samples where cytogenetic risk could not be determined "N.D":
```{r}
# CRGIP = Cytogenetic Response Good + Intermediate + Poor
rDesCRGIP <- droplevels(subset(rDes, Cytogenetic_risk != "N.D."))
str(rDesCRGIP)
dim(rDesCRGIP)
rDatCRGIP <- rDat[ , names(rDat) %in% rDesCRGIP$TCGA_patient_id]
dim(rDatCRGIP)
identical(names(rDatCRGIP), as.character(rDesCRGIP$TCGA_patient_id))
cytoRisk <- rDesCRGIP$Cytogenetic_risk
```

I will make a model with a reference + treatment effect, where I will use "Good" risk as the intercept:
```{r}
normFactor <- calcNormFactors(rDatCRGIP)
design <- model.matrix(~ cytoRisk)
colnames(design)
# The intercept represents "Good" cytogenetic risk
head(design)
rDatCRGIPvoom <- voom(rDatCRGIP, design, plot = TRUE, 
                      lib.size = colSums(rDatCRGIP) * normFactor)
fit <- lmFit(rDatCRGIPvoom, design)
fit <- eBayes(fit)
voomCR <- topTable(fit, coef = c("cytoRiskIntermediate", "cytoRiskPoor"), 
                   p.value = 1e-5, n = Inf)
nrow(voomCR)
head(voomCR)
voomCRgenes <- rownames(voomCR)
```

```{r include=FALSE}
# Genes differentially expressed between Good and Intermediate or Poor cytogenetic risk:
dgeGlm <- DGEList(counts = rDatCRGIP, group = cytoRisk)
plotSmear(dgeGlm, de.tags = voomCRgenes)
abline(h = c(-1, 1), col = "blue")
# Ok this doesn't look right. There are genes coloured red with very little logFC? But hang on, plotSmear plots logCF between two experimental groups. It has chosen to plot Good against intermediate by default??? Confusing. I don't think you can use this plot when we are comparing 3 groups between each other.
```

Genes differentially expressed between Good and Poor cytogenetic risk:
```{r}
voomCRGP <- topTable(fit, coef = "cytoRiskPoor", p.value = 1e-5, n = Inf)
nrow(voomCRGP)
head(voomCRGP)
```

Plot top genes differentially expressed between Good and Poor cytogenetic risk:
```{r}
voomCRGPgenes <- rownames(voomCRGP)
rDatvoomCR <- rDatCRGIP[head(voomCRGPgenes), ]
dim(rDatvoomCR)
rDatvoomCR$Transcript <- rownames(rDatvoomCR)
rDatvoomCR <- melt(rDatvoomCR, id.vars = "Transcript", 
                   variable.name = "TCGA_patient_id",
                   value.name = "Read_counts")
rDatvoomCR$Transcript <- gsub("[|].*$", "", rDatvoomCR$Transcript)
head(rDatvoomCR)
rDatvoomCR <- merge(rDatvoomCR, rDesCRGIP, by = "TCGA_patient_id")
head(rDatvoomCR)
ggplot(rDatvoomCR, aes(Transcript, log2(Read_counts), 
                       colour = Cytogenetic_risk)) +
  geom_boxplot() +
  facet_wrap(~ Transcript, scales = "free") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())  
```

Genes differentially expressed between Good and Intermediate cytogenetic risk:
```{r}
voomCRGI <- topTable(fit, coef = "cytoRiskIntermediate", p.value = 1e-5, 
                     n = Inf)
nrow(voomCRGI)
head(voomCRGI)
```

Plot top genes differentially expressed between Good and Intermediate cytogenetic risk:
```{r}
voomCRGIgenes <- rownames(voomCRGI)
rDatvoomCR <- rDatCRGIP[head(voomCRGIgenes), ]
dim(rDatvoomCR)
rDatvoomCR$Transcript <- rownames(rDatvoomCR)
rDatvoomCR <- melt(rDatvoomCR, id.vars = "Transcript", 
                   variable.name = "TCGA_patient_id",
                   value.name = "Read_counts")
rDatvoomCR$Transcript <- gsub("[|].*$", "", rDatvoomCR$Transcript)
head(rDatvoomCR)
rDatvoomCR <- merge(rDatvoomCR, rDesCRGIP, by = "TCGA_patient_id")
head(rDatvoomCR)
ggplot(rDatvoomCR, aes(Transcript, log2(Read_counts), 
                       colour = Cytogenetic_risk)) +
  geom_boxplot() +
  facet_wrap(~ Transcript, scales = "free") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
```

How can we compare Intermediate and Poor risk? Create a contrast matrix:
```{r}
colnames(design)
(contMatrix <- makeContrasts(
  PoorVsInt = cytoRiskPoor - cytoRiskIntermediate,
  levels = design))
fitCont <- contrasts.fit(fit, contMatrix)
EbFitCont <- eBayes(fitCont)
voomCRIP <- topTable(EbFitCont, p.value = 1e-5, n = Inf)
nrow(voomCRIP)
head(voomCRIP)
resCont <- decideTests(EbFitCont, p.value = 1e-5, method = "global")
summary(resCont)
# negHits <- rownames(rDatCRGIP)[which(resCont[ , "PoorVsInt"] < 0)]
# posHits <- rownames(rDatCRGIP)[which(resCont[ , "PoorVsInt"] > 0)]
```

Plot top genes differentially expressed between Intermediate and Poor cytogenetic risk:
```{r}
voomCRIPgenes <- rownames(voomCRIP)
rDatvoomCR <- rDatCRGIP[head(voomCRIPgenes), ]
dim(rDatvoomCR)
rDatvoomCR$Transcript <- rownames(rDatvoomCR)
rDatvoomCR <- melt(rDatvoomCR, id.vars = "Transcript", 
                   variable.name = "TCGA_patient_id",
                   value.name = "Read_counts")
rDatvoomCR$Transcript <- gsub("[|].*$", "", rDatvoomCR$Transcript)
head(rDatvoomCR)
rDatvoomCR <- merge(rDatvoomCR, rDesCRGIP, by = "TCGA_patient_id")
head(rDatvoomCR)
ggplot(rDatvoomCR, aes(Transcript, log2(Read_counts), 
                       colour = Cytogenetic_risk)) +
  geom_boxplot() +
  facet_wrap(~ Transcript, scales = "free") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
```

Create a Venn diagram to show which genes were differentially expressed:
```{r}
deGenes <- list(IntermedVsGood = voomCRGIgenes, 
                PoorVsIntermed = voomCRIPgenes, 
                PoorVsGood = voomCRGPgenes)
plot.new()
vennPlot <- venn.diagram(deGenes, filename = NULL, force.unique = TRUE, 
                         fill = c("red", "blue", "yellow"))
grid.draw(vennPlot)
```

Wow one gene is differentially expressed between all groups? What is it?!
```{r}
(luckyGene <- (intersect(intersect(voomCRGIgenes, voomCRIPgenes), 
                        voomCRGPgenes)))
```

It is a zinc finger protein on chromosome 7, which we know is commonly deleted in AML! Nice. Let's check its expression!
```{r}
rDatvoomCR <- rDatCRGIP[head(luckyGene), ]
rDatvoomCR$Transcript <- rownames(rDatvoomCR)
rDatvoomCR <- melt(rDatvoomCR, id.vars = "Transcript", 
                   variable.name = "TCGA_patient_id",
                   value.name = "Read_counts")
rDatvoomCR$Transcript <- gsub("[|].*$", "", rDatvoomCR$Transcript)
head(rDatvoomCR)
rDatvoomCR <- merge(rDatvoomCR, rDesCRGIP, by = "TCGA_patient_id")
head(rDatvoomCR)
ggplot(rDatvoomCR, aes(Transcript, log2(Read_counts), 
                       colour = Cytogenetic_risk)) +
  geom_boxplot()
```
