Clinical Data Import and Cleaning
=================================

> To knit .rmd file, make sure the working directory is set to that of the source file  

Load required libraries and print working directory:
```{r message=FALSE}
library(RCurl)
library(knitr)
```

Read in the clinical data-sheet from the [TCGA Publication Website](https://tcga-data.nci.nih.gov/docs/publications/laml_2012/):

```{r tidy=FALSE}
# To re-download data from TCGA portal site, uncomment this block
# Require "ssl.verifypeer = FALSE" to avoid SSL certificate issues
#url <- "https://tcga-data.nci.nih.gov/docs/publications/laml_2012/clinical_patient_laml.tsv"
#raw_clinical_data <- getURL(url, ssl.verifypeer = FALSE)
#raw_clinical_data <- read.table(text = raw_clinical_data, header = TRUE, sep ='\t')
# Read in raw clinical data:
raw_clinical_data <- read.table("../../data/clinical_patient_laml.tsv", header = TRUE, 
                                 sep = "\t")
```

```{r}
str(raw_clinical_data, max.level = 0)
names(raw_clinical_data)
```
This data-sheet has `r length(raw_clinical_data)` columns, a lot more variables than we'll need for this project. The `r nrow(raw_clinical_data)` rows represent the patients.

Also available, as an Excel sheet, is a Supplementary table listing much of the same information, but with more annotation.

*Note:* We're using the updated version as of 2013-05-13. Since the data is only available as an Excel sheet, we've saved the data in .csv format outside of R:*

- Download the Supplementary file from [the paper website](https://tcga-data.nci.nih.gov/docs/publications/laml_2012/SuppTable01.update.2013.05.13.xlsx)
- Open in Excel, save as CSV

```{r}
supp_d <- read.csv("../../data/SuppTable01.update.2013.05.13.csv")
str(supp_d, max.level = 0)
```

As a first pass, we'll try to clean the `cytogenetic_abnormality` column from the raw clinical data into a simpler categorical variable:

```{r}
summary(raw_clinical_data$cytogenetic_abnormality)
```

There are some immediate problems with this approach:

- `r sum(raw_clinical_data$cytogenetic_abnormality == "[Not Available]")` samples are missing data
- Several samples have many of the CNAs of interest
- The notation seems inconsistent between samples

Alternate approaches possible:

1. Limit the analysis to things with a single, distinct karyotypic event (will lose lots of samples)
2. Make a 'best guess' at the most relevant alteration for each sample (messy)
3. Make several different binary variables for each of the main CNAs, e.g.:

| Patient | del(5q) | del(7q) | Trisomy 8 | t(15:17) | inv(16) | 
| ------- | ------- | ------- | --------- | -------- | ------- |
| 1 | T | F | F | F |T |
| 2 | F | F | F |T | F |

Thoughts? This would make the annotation simpler, but might complicate the analysis a bit.

### Follow-up to that question: 

#### Summarize Columns in Raw Clinical Data and Supplementary Table 1

To address the comments at [issue #3](https://github.com/rdocking/stat540-group-project-aml-cnv/issues/3), I'm going to try a few approaches to stratifying the data-set.

First, I'll work on trying to get a workable stratification out of the `raw_clinical_data` imported above. The main issues I've been having so far:

- In the `cytogenetic_abnormality` column, the entries are sometimes split by pipes, e.g. 'Normal|del (7q) / 7q-' vs. 'del (7q) / 7q-' - it's not clear that these should be treated equivalently.
- There is a second column, `cytogenetic_abnormality_other` that is even more inconsistently filled in (i.e. no data for 172 samples, uses 'no', 'No', and 'NO' for others)

This is looking a bit like a rabbit-hole. I'm going to try another tack and see if I can regenerate the sample counts from Table 1 in the main text of the paper:

![Table 1 Snippet](table_1_snippet.png)

From Supplementary Table 1, (loaded above as `supp_d`), the cytogenetic data is available in the following columns:

- `Cytogenetics` - karyotype using [cytogenetic nomenclature](http://www.radford.edu/~rsheehy/cytogenetics/Cytogenetic_Nomeclature.html). These seem consistently filled in (yay!), but are tricky to parse (boo!)
- `Gene Fusions by RNA-Seq` - predicted gene fusions from the RNA-Seq data. This has entries like 'PML(+)RARA(+) (In frame)', which corresponds to 't(15;17)' from the `Cytogenetics` column. This _should_ be pretty accurate for fusions, but not for CNAs
- `Inferred genomic rearrangement (from RNA-Seq fusion)` - translation of `Gene Fusions by RNA-Seq` into `Cytogenetics` nomenclature
- `Cytogenetic Classification` - A categorical variable summarizing the previous columns:

```{r}
summary(supp_d$Cytogenetic.Classification)
```

  this is _not quite_ what we're after - we'd like to call individual karyotype-level events.
- `RISK (Cyto)` - another categorical classification of the samples by risk type:

```{r}
summary(supp_d$RISK..Cyto.)
```

- Then two more classifications, `Molecular Classification` and `RISK (Molecular)`
  
#### Annotate Samples with Karyotypic events of interest  

Per conversation with group members and Shaun, I'll clean the supplemental data in `SuppTable01.update.2013.05.13.csv`. Note that I'm doing this in Python with [`parse_supplementary_table.py`](https://github.com/rdocking/stat540-group-project-aml-cnv/blob/master/code/parse_supplementary_table.py) for expediency.

Manipulations are as follows:

- Select a relevant subset of variables
- For variables that can be converted to Boolean types, do so
- Convert column names to be more R-friendly (e.g., `'Expired?  4.30.13'` became `Expired` and `'%BM Blast'` became `BM_blast_pct`)

The script should be run like so:

```
python parse_supplementary_table.py > experimental_design.csv
```

Now I can read in the modified CSV file to get a cleaner data frame:
```{r}
cleaned_data <- read.csv("../../data/experimental_design.csv")
```

This data frame is quite a bit smaller and simpler than the original:
```{r}
str(cleaned_data)
```

Here is a reduced data frame, with just the main karyotypic variables of interest:
```{r}
karyotypes <- subset(cleaned_data, select = c('trisomy_8', 'del_5', 'del_7'))
summary(karyotypes)
```

*Note:* In the original version of the parsing script, I selected for 'del(5q)' and 'del(7q)'. In the current version, the selection is expanded to include full-chromosome losses as well as the smaller deletions. The revised version gives numbers that more accurately match the numbers presented in the paper.

Summarizing the different classes:
```{r}
normal_karyotype <- nrow(subset(karyotypes, trisomy_8 == FALSE & del_5 == FALSE & del_7 == FALSE))
trisomy_8_exclusive <- nrow(subset(karyotypes, trisomy_8 == TRUE & del_5 == FALSE & del_7 == FALSE))
del_5_exclusive <- nrow(subset(karyotypes, trisomy_8 == FALSE & del_5 == TRUE & del_7 == FALSE))
del_7_exclusive <- nrow(subset(karyotypes, trisomy_8 == FALSE & del_5 == FALSE & del_7 == TRUE))
t8_del5 <- nrow(subset(karyotypes, trisomy_8 == TRUE & del_5 == TRUE & del_7 == FALSE)) 
t8_del7 <- nrow(subset(karyotypes, trisomy_8 == TRUE & del_5 == FALSE & del_7 == TRUE))
del5_del7 <- nrow(subset(karyotypes, trisomy_8 == FALSE & del_5 == TRUE & del_7 == TRUE))
all_three <- nrow(subset(karyotypes, trisomy_8 == TRUE & del_5 == TRUE & del_7 == TRUE))
count <- c(normal_karyotype, trisomy_8_exclusive, del_5_exclusive, del_7_exclusive, t8_del5, t8_del7,  del5_del7, all_three)
summary_frame <- data.frame(count, row.names = c('Normal Karyotype', 'Trisomy 8', 'Loss of 5 or del(5q)', 'Loss of 7 or del(7q)', '+8, -5', '+8, -7', '-5, -7', '+8, -5, -7'))
```

Print the summary table:
```{r results='asis'}
kable(summary_frame, format='markdown')
```

So, we have a few cases where the abnormalities of interest occur in isolation, but several where they overlap.  


### Further cleaning of the experimental design file `cleaned_dat`

We only need to keep the patients for which we have RNA-seq data available:
```{r}
cleaned_data <- cleaned_data[cleaned_data$RNAseq_available == TRUE, ]
dim(cleaned_data)
```

We won't need all the variables stored in the experimental design file, so let's keep the most relevant variables for our analyses:

```{r}
cleanExpDes <- cleaned_data[ , c("TCGA_patient_id", "Sex", "Race", 
                                 "FAB_subtype", "Age", "trisomy_8", "del_5",
                                  "del_7", "Cytogenetic_risk", 
                                 "Molecular_risk")]
str(cleanExpDes)
```

Much better, now we only have `r length(cleanExpDes)` variables (columns) instead of `r length(cleaned_data)` variables.

Order the rows in 'cleanExpDes' by TCGA Patient ID:
```{r}
cleanExpDes <- cleanExpDes[order(cleanExpDes$TCGA_patient_id), ]
head(cleanExpDes)
```

Save the output to file:
```{r}
write.table(cleanExpDes, "../../data/experimental_design_cleaned.txt", sep = "\t",
            row.names = FALSE)
```

Ensure we can read the file back in correctly:
```{r}
test <- read.delim("../../data/experimental_design_cleaned.txt")
str(test)
head(test)
tail(test)
```

Great, so we will use the `experimental_design_cleaned.txt` file our experimental design, where the `r nrow(cleanExpDes)` rows represent the patients for which we have RNA-seq data available, and are sorted by their TCGA patient ID. There are `r length(cleanExpDes)` columns which represent the key variables we want to include in our linear modelling and machine learning tasks.
